.github/workflows/image.yaml

```diff
@@ -29,6 +29,52 @@ on:
       - release-*
 
 jobs:
+  gh-runner-image:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        name: Check out code
+      - name: Calculate build vars
+        id: vars
+        run: |
+          if [ "${{github.event_name == 'pull_request'}}" = "true" ]; then
+            echo "IMAGE_PUSH=false" >> $GITHUB_ENV
+            echo "IMAGE_OUTPUTS=type=docker,dest=/tmp/k8s-test-infra-image.tar" >> $GITHUB_ENV
+            echo "IMAGE_PLATFORMS=linux/amd64" >> $GITHUB_ENV
+          else
+            echo "IMAGE_PUSH=true" >> $GITHUB_ENV
+            echo "IMAGE_OUTPUTS=type=registry,push=true" >> $GITHUB_ENV
+            echo "IMAGE_PLATFORMS=linux/amd64,linux/arm64" >> $GITHUB_ENV
+          fi
+          HOLODECK_VERSION=$(sed -n 's/.*holodeck:v\([0-9.]*\).*/\1/p' images/gh_runner/Dockerfile)
```

Comment:
I think we should just define the versions as constants in `*.mk` or `Makefile` files and pass them as build args to the docker containers

Reply:
We want to rely on dependabot